### ISSUE TITLE
Boosted Regression Tree Model Refinement

### ISSUE THREAD

**Issue by orhuna** [orhuna/ar_ivt#4]
Defining the Boosted Regression Tree Model with best bias-variance tradeoff.

**Closing Criteria**
- [x] Optimal XGBoost Model parameters
- [x] Pushing the code for optimizing the XGBoost model

**Comment by orhuna** [comment]
@kellyrmat take a look at [AutoML for XGBoost](https://microsoft.github.io/FLAML/docs/Examples/AutoML-for-XGBoost/).

It would be time effective to use hyperparameter + model optimizers such as this going forward.

**Comment by orhuna** [comment]
@kellyrmat looks promising! What's the accuracy of the optimal model?

**Comment by kellyrmat** [comment]
@orhuna My best r^2 on the testing dataset has been 0.8026. So, not bad!

**Comment by orhuna** [comment]
@kellyrmat not bad indeed but it can be better 🚀 

Please post the snippet for the boruta wrapper, there can be objective functions that will favor a high test R2 and optimize the model.

**Comment by kellyrmat** [comment]
@orhuna Here's a snippet of the hyperparameters I've determined are 'optimal' by using the AutoML tuner. I tuned hyperparameters with AutoML and then made an XGBregressor with those params. Then, I did a boruta search using the XGBregressor. 

![image](https://github.com/orhuna/ar_ivt/assets/114206053/6834e7e5-66e9-4ec9-81c3-d047dcf31c78)

**Comment by kellyrmat** [comment]
@orhuna I'll modify the train importance to False to favor the testing r2. Also, I just realized I can supply a parameter grid here to do a simultaneous hyperparameter search. I was overthinking it. 
![image](https://github.com/orhuna/ar_ivt/assets/114206053/38d58f14-a47d-408d-bc15-dde2be7c7db2)

**Comment by orhuna** [comment]
Specs satisfied. @kellyrmat ready to close.
